### PR TITLE
chore: fe deployment

### PR DESCRIPTION
**Pull request by bogdan-manole**
### Summary by CodeRabbit

# Release Notes

**Refactor:**
- Updated Dockerfile.web to improve build process and environment variable handling.
- Modified docker-files/run-server.sh and docker-files/run-web.sh for better Docker image execution.
- Enhanced flake.nix with new imports to include both server and web applications.
- Adjusted react-web/.env.production environment variables for updated configuration.

**New Feature:**
- Introduced new functions in web.nix for building, running, and pushing Docker images. 

These changes aim to streamline the Docker image creation and deployment process, enhancing the overall performance and maintainability of the application.

### CODE REVIEW

File: Dockerfile.web [hunk 1]
```diff
@@ -2,17 +2,23 @@
...
+
+RUN npm run build
 
 COPY ./docker-files/start_web.sh .
 RUN chmod +x ./start_web.sh
 
 ENTRYPOINT [ "./start_web.sh" ]
 #ENTRYPOINT [ "/bin/sh","-c","./start_web.sh"]
-CMD [ "npm","start" ]
+#CMD [ "npm","start" ]
```
[the remark this answers is not in the document]
Comment (RSoulatIOHK) [reply Dockerfile.web:24]: @coderabbitai Can you explain more?

File: Dockerfile.web [hunk 2]
```diff
@@ -2,17 +2,23 @@ FROM node:16
 
 WORKDIR /app/
 COPY react-web /app
-COPY ./docker-files/.env.local .
-RUN npm install
+#COPY ./docker-files/.env.local .
+RUN npm install --global serve
```
[the remark this answers is not in the document]
Comment (bogdan-manole) [reply Dockerfile.web:6]: not bad 😄

File: Dockerfile.web (hunk 1 above)
[the remark this answers is not in the document]
Comment (bogdan-manole) [reply Dockerfile.web:24]: @coderabbit `server build` command from the last line of the entry point file starts the application from the `build` folder

**Comment by bogdan-manole** [comment]
the fe was detached from this repo

**Pull request closed without merging**